core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -1286,12 +1286,12 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
         rebuildAppStore(store, reader, attempt.info.lastUpdated.getTime())
         hybridStore = store
       } catch {
-        case _: IOException if !retried =>
+        case ioe: IOException if !retried =>
           // compaction may touch the file(s) which app rebuild wants to read
           // compaction wouldn't run in short interval, so try again...
           logWarning(log"Exception occurred while rebuilding log path " +
             log"${MDC(PATH, attempt.logPath)} - " +
-            log"trying again...")
+            log"trying again...", ioe)
```

Comment:
Do we want to log this as warming ?
INFO for exceptions stack trace  is more reasonable - it is more developer facing.

Reply:
Warning feels right to me,  but I'm open to changing it. My mental model for log levels is typically:

Info -> Developer facing that describes what the system is doing
Warning -> Something went wrong, but it's not fatal and it will try to keep going
Error -> Something went wrong and isn't recoverable or will at least result in degraded functionality

Reply:
I dont have a strong opinion in this specific case.

This is a retriable failure, and could simply succeed on next attempt. Note that the exception, when if it gets bubbled up (`retried == true`), gets logged at `INFO` currently (see `loadDiskStore`) - so the `logWarning` level here, to begin with, looks slightly off.

+CC @HeartSaVioR who reviewed the original PR

Reply:
I agree with @robreeves about the mental model for logging level, but also agree with @mridulm about the inconsistency of logging level. From this method by itself, my 2 cents goes to WARN. But in the whole context, this method is called only from loadDiskStore and it logs the non-retriable (or already retried) exception as INFO, because it's not end of the world and there will be another fallback to RocksDB.

If we prefer consistency then maybe INFO is better, but then this logging level is strongly coupled with the caller's context, so it's like a trade-off / preference. Maybe fallback attempts may worth to log warn - it is loadDiskStore we need to change to WARN? I don't have a strong preference.

Reply:
Thanks for the details @HeartSaVioR !
IMO we should change one of them - either change info -> warn, or (and I marginally prefer this) change the warn -> info ...

Reply:
I'm OK to change this to INFO as there will be fallback anyway. @robreeves Would you mind making a change, or would you mind keeping the level for specific reason?

Reply:
Sure, updated to an info log.